### ISSUE TITLE
Decimal precision truncated when serialize/deserialize std::vector<double> container

### ISSUE THREAD

**Issue by zhoupengwei** [biojppm/rapidyaml#417]
Hi author, I found that the std::vector<double> lost precision during the process of deserializing, As the following sample code:
```c++
ryml::Tree tree;

  {
    std::vector<double> test_vec{1.23234412342131234,
                                 2.12323123143434237,
                                 3.67847983572591234};
    ryml::NodeRef root = tree.rootref();
    root |= ryml::SEQ;
    root << test_vec;
  }

  {
    ryml::ConstNodeRef root = tree.crootref();
    std::vector<double> test_vec;
    root >> test_vec;
    for (const auto& elem : test_vec) {
      std::cout << "elem " << elem << std::endl;
    }
  }
```
The program will output:
elem 1.23234
elem 2.12323
elem 3.67848

How can I avoid losing numerical accuracy ？

**Comment by biojppm** [comment]
TL;DR:
```c++
        ryml::Tree serialized;
        ryml::NodeRef root = serialized.rootref();
        root |= ryml::SEQ;
        for(const double v : test_vec)
            root.append_child() << ryml::fmt::real(v, 17, ryml::FTOA_FLOAT);
```
For more details see the linked PR, which adds an example and explanation to the quickstart.

**Comment by zhoupengwei** [comment]
@biojppm Thank you for providing detailed examples and great project.
